src/code/providers/google-drive-provider.ts

```diff
@@ -226,50 +248,50 @@ class GoogleDriveProvider extends ProviderInterface {
     })
   }
 
-  remove(metadata, callback) {
+  remove(metadata: CloudMetadata, callback?: ProviderRemoveCallback) {
     return this._waitForGAPILoad().then(() => {
       const request = gapi.client.drive.files["delete"]({
         fileId: metadata.providerData.id})
-      return request.execute(result => typeof callback === 'function' ? callback((result != null ? result.error : undefined) || null) : undefined)
+      return request.execute((result: any) => callback?.(result?.error))
     })
   }
 
-  rename(metadata, newName, callback) {
+  rename(metadata: CloudMetadata, newName: string, callback?: (err: string | null, metadata?: CloudMetadata) => void) {
     return this._waitForGAPILoad().then(() => {
```

Comment:
Not directly related to this change, but the _waitForGAPILoad() function looks like this:
```typescript
  _waitForGAPILoad() {
    return new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = "https://apis.google.com/js/api.js"
      script.onload = () => {
        gapi.load("client:auth2", () => {
          gapi.client.init({
            apiKey: this.apiKey,
            clientId: this.clientId,
            discoveryDocs: ["https://www.googleapis.com/discovery/v1/apis/drive/v3/rest"],
            scope: this.scopes
          })
          .then(resolve)
          .catch(reject)  // eslint-disable-line @typescript-eslint/dot-notation
        })
      }
      document.head.appendChild(script)
    })
  }
```
Unless I'm missing something, that appears to be loading the API every time it's called. Given that this function is called by nearly every other function in this module, that seems to suggest that we're loading the API more than necessary. I would have expected to see a check for whether the API is already loaded and an immediate return if detected.

Reply:
I think you are correct.  We should have a class member to act as a singleton holding the Promise and do something like:

```
return this.loadPromise || (this.loadPromise = return new Promise(...
```